### DISCUSSION TITLE
Reconnect Behavior for Edges

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#1434] [Q&A] [answered]
I have a question about reconnecting edges:

When I try to reconnect an existing Edge in GLSP with another element in my diagram the behavior is a little bit strange and users often do not understand it.

![Peek 2024-12-03 19-42](https://github.com/user-attachments/assets/4c79e7a5-49b8-440d-8b38-cd3f2b1bafcb)

1. I need first click on the target waypoint of the edge, 
2. now I can move the waypoint around in the diagram,
3. but I can't drop it on an element
4. first, I have to release the mouse button
5. than I can click on the new target element

This releasing of the mouse button can not be seen in the screen cast above. But I wonder if this is the normal expected behavior of reconnecting edges in glsp?

The more expected behavior for me would be that releasing the mouse button finishes the reconnect action. Maybe I have done something wrong in my implementation?

===
Ralph

**Comment by tortmayr** [comment, marked as answer]
Hi @rsoika,

I think that`s the normal/expected behavior of the default reconnect tool.
It`s currently implemented in the same fashion as the edge creation tool which also does not really support the drag usecase.
Currently these tools are designed to be used in `click-move-click` style:
Edge creation:
-  Select the source element (click)
-  Move/hover of the intended target (move)
-  Select target (click)

Edge reconnect:
- Select reconnect handle (click)
- Move/hover of the intended target (move)
- Select target (click)

I agree that this is a bit unexpected and especially for the reconnect use-case it would be great if the tool would also allow reconnection via `drag&drop`

Feel free to open an issue//feature request for this

**Reply by rsoika** [reply]
OK, understood. If this is fine for you it's fine for me. I just was not sure if I missed something.

_click-move-click style_ is easy enough to explain. And this reminds me that I should write a basic manual for my tool ;-)
